pkg/types/core/median.go

```diff
@@ -2,15 +2,23 @@ package core
 
 import (
 	"context"
+	"encoding/json"
 
 	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
 
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink-common/pkg/services"
 	"github.com/smartcontractkit/chainlink-common/pkg/types"
 )
 
+type DeviationFuncDefinition interface {
+	Func(lggr logger.Logger) median.DeviationFunc
+	json.Marshaler
+	json.Unmarshaler
+}
```

Comment:
I think this should be a pure message type. Different funcs can be called dynamically depending on what data is in the message, but merging the two as a single type creates complexity.

Reply:
`NewMedianFactory` already takes a bunch of interface args and stuff tho?

Reply:
Every interface requires a client/server pair. Is that really necessary in this case?